docs/source/integration-guide.rst

```diff
@@ -0,0 +1,189 @@
+
+
+Wrapping native workload executors with a shell script
+------------------------------------------------------
+
+It might be difficult for some driver languages to create a single executable file for the workload executor as
+required by ``astrolabe``. In such cases, drivers may choose to wrap the actual call to a natively implemented
+workload executor in a shell script such that exposes the API desired by the
+:ref:`workload-executor-specification` specification.
+
+.. note::
+
+   PyMongo's ``astrolabe`` integration uses this pattern to implement its
+   `workload executor <https://github.com/mongodb-labs/drivers-atlas-testing/blob/master/.evergreen/python/pymongo/workload-executor.sh>`_.
+   Check it out for a practical example of what this looks like.
+
+Certain special considerations apply to drivers that choose to implement their workload executors in this manner:
+
+* The shell script used to wrap the native workload executor implementation MUST ensure that it traps the
+  ``INT`` signal and sets a zero exit-code to prevent the Test Orchestrator from misconstruing all interruptions
+  of the native workload executor as failures (shell scripts will set always set a non-zero exit-code when they
+  terminate after encountering the ``INT`` signal). To do this, add the following snippet to your script *before*
+  the call to your native workload executor::
+
+    trap "exit 0" INT
+
+* The shell script MUST NOT call the native workload executor as a background process (i.e. using ``&`` control
+  operator). Doing so will cause the native workload executor to run in the background in a subshell and make it
+  impossible for the Test Orchestrator to capture the STDOUT and STDERR streams of the workload executor.
+
+
+----------------------------------------------
+Step 2: Implementing a Driver Installer Script
+----------------------------------------------
+
+Drivers must implement this standalone script to perform all setup/installation-related tasks for their driver.
+The installer script MUST be saved in the ``DRIVER_DIRNAME`` directory under the name ``install-driver.sh``.
+The executable permission bit MUST be set for the install script file before it is committed to git.
+
+This script can be used to perform any number of arbitrary tasks related to setting up the environment for
+the workload executor to be executed within. It MUST NOT however, clone the driver source repository.
+
+.. note::
+
+   All Evergreen expansions for a given build variant are available to the installer script
+   at runtime as environment variables. Scripts can make use of any of these environment variables but must
+   ensure that they are written in a way that prevents accidentally leaking Atlas API credentials and
+   MongoDB user credentials.
+
+
+.. note::
+
+   Even if a driver does not require an installer script, it MUST create an empty/no-op script of the required
```

Comment:
I think we can probably work around this in the evergreen configuration? Just test for the existence of the file and only run it if it is present?

Reply:
Good point. Done.